### PR TITLE
EMP-310: Complex page style formatting

### PR DESCRIPTION
**Pull request by muyinatech**
**Changes made:**
- Added date-fns library for date & time formatting
- Added FieldType type
- Added type property to ConfigField & DisplayField types
- Updated CRM5 display config to specify currency, date, time types for the required fields
- Moved display config validation to a separate file
- Added additional display config schema validation tests
- Created CRM Field formatter for format fields for the given type
- Added nunjucks filters for formatting
- Create formField template macro

### CODE REVIEW

File: server/services/config/crm5DisplayConfig.json [hunk 1]
```diff
@@ -464,11 +481,13 @@
           "fields": [
             {
               "label": "Attendance Time",
-              "apiField": "AllCosts.AccruedCosts.Attendance.time"
+              "apiField": "AllCosts.AccruedCosts.Attendance.time",
```
Comment (muyinatech) [inline server/services/config/crm5DisplayConfig.json:484]: Same explanation as below

File: server/utils/crmFieldFormatter.test.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    it('should not format currency if not a number', () => {
+      const result = formatCurrency('blah')
+      expect(result).toEqual('blah')
+    })
+  })
+
+  describe('formatDate', () => {
+    it('should format date', () => {
+      const result = formatDate('2024-02-14T00:00:00.000+00:00')
+      expect(result).toEqual('14 02 2024')
```
Comment (sarahyjja) [inline server/utils/crmFieldFormatter.test.ts:24]: The date format should not retrieve the `0` in day and month
For eg.: 14 2 2024

File: server/utils/crmFieldFormatter.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+export const formatCurrency = (value: string): string => {
+  const number = Number(value)
+  if (Number.isNaN(number)) return value
+  return currencyFormat.format(number)
+}
+
+export const formatDate = (value: string): string => {
+  const date = Date.parse(value)
+  if (Number.isNaN(date)) return value
+  return format(date, 'dd MM yyyy')
```
Comment (sarahyjja) [inline server/utils/crmFieldFormatter.ts:17]: Docs about not retrieving the `0` : https://date-fns.org/v3.6.0/docs/intlFormat

**Review by sarahyjja — CHANGES_REQUESTED** [review]
Just a minor change about not retrieving the `0` in `date` format (posted a link to the intlFormat). 
Out of that it LGTM!

**Review by sarahyjja — APPROVED** [review]
LGTM

**Pull request merged**